pytorch_lightning/plugins/training_type/training_type_plugin.py

```diff
@@ -425,6 +443,7 @@ def teardown(self) -> None:
 
         It is the right place to release memory and free other resources.
         """
+        self._move_optimizer_state(torch.device("cpu"))
```

Comment:
Note, if I read correctly this was previously only done for the GPU accelerator. Should be fine, just pointing it out in case.

Reply:
I don't remember you or me did this. But probably we should keep this in GPUAccelerator teardown. How about we address it together in https://github.com/PyTorchLightning/pytorch-lightning/issues/11091?

Reply:
I don't remember you or me did this. But probably we should keep this in GPUAccelerator teardown. How about we address it together in #11091?

Reply:
should be good either way. for tpu it's just redundant because the teardown runs in the spawned processes which, when they end, get completely destroyed so nothing is left in memory. 
But in the future, we may decide to bring the optimizer states back to the main process through the queue and restore them like we  do for the model weights. Anyway, this is open for discussion